### ISSUE TITLE
Excess characters in line

### ISSUE THREAD

**Issue by willmcgugan** [Erotemic/progiter#21]
**Describe the bug**

When the length of the progress information reduces, it leaves excess characters on the line. Note the cursor in the following screenshot:

<img width="686" alt="Screenshot 2022-08-27 at 08 09 29" src="https://user-images.githubusercontent.com/554369/187019357-2336ebfc-49d0-42ba-8139-767daad3bfaa.png">

**To Reproduce**
```python
from random import randint
from time import time, sleep
import progiter

start = time()
for i in progiter.ProgIter(range(10000)):
    sleep(i/1000000)
print(time() - start)
```

**Expected behavior**
The excess characters should be cleared.

**Desktop (please complete the following information):**

MacOS
progiter 0.1.4
Python 3.10.4

**Comment by Erotemic** [comment]
I've been aware of this bug for awhile, and I did attempt to fix it at one point, but I got stuck and it wasn't prevalent where I ever came back to it. But I imagine you must be able to do something similar in rich?

The following code breaks down your MWE:

```python
import ubelt as ub
cap = ub.CaptureStdout()

with cap:
    from random import randint
    from time import time, sleep
    import progiter

    start = time()
    for i in progiter.ProgIter(range(100)):
        sleep(i/10000)
    print(time() - start)

import re
import sys
parts = re.split('(\r)', cap.text)
print('parts = {}'.format(ub.repr2(parts, nl=1)))
```

The sequence of characters written to stdout is:

```python
parts = [
    '',
    '\r',
    '   0/100... rate=0 Hz, eta=?, total=0:00:00',
    '\r',
    '   1/100... rate=113147.77 Hz, eta=0:00:00, total=0:00:00',
    '\r',
    '   4/100... rate=5107.76 Hz, eta=0:00:00, total=0:00:00',
    '\r',
    '  16/100... rate=1225.94 Hz, eta=0:00:00, total=0:00:00',
    '\r',
    '  64/100... rate=306.43 Hz, eta=0:00:00, total=0:00:00',
    '\r',
    ' 100/100... rate=197.16 Hz, eta=0:00:00, total=0:00:00\n0.5073776245117188\n',
]
```

for my reference, this slowly steps through this sequence:

```python
print(f'Slowly show progression of bug: ({len(parts)} steps)')
for p in parts:
    sleep(1)
    sys.stdout.write(p)
    sys.stdout.flush()
```

The `\r` character instructs the tty to clear all characters before it, but nothing after it. I think I've attempted to use `\33[2K`  as detailed by [this SO post](https://stackoverflow.com/questions/1508490/erase-the-current-printed-console-line), but I recall it not working reliably. Maybe that is a false memory though.

Do you have any recommendations on the best way to clear the current line? Is there a place in rich that demonstrate how to do this right?

**Comment by willmcgugan** [comment]
I think the most reliable way to do this would be to pad with spaces to overwrite any excess. When you clear the line you risk introducing flickering.

**Comment by Erotemic** [comment]
That makes sense. I have an initial patch in https://github.com/Erotemic/progiter/pull/22 which I'll need to cleanup to make the tests pass. I think there are unhandled ansi escape code / unicode corner cases, but the changes should be strict improvement.

Thanks for the report and the advice!

**Comment by Erotemic** [comment]
The issue has been fixed with the merge of #23. Currently running the CI to publish the new version with the fix to pypi.
